query/src/servers/http/formats/tsv_output.rs

```diff
@@ -0,0 +1,52 @@
+// Copyright 2022 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use common_datablocks::DataBlock;
+use common_exception::ErrorCode;
+use common_exception::Result;
+
+const FIELD_DELIMITER: u8 = b'\t';
+const ROW_DELIMITER: u8 = b'\n';
+
+pub fn block_to_tsv(block: &DataBlock) -> Result<Vec<u8>> {
```

Comment:
Maybe we should move formats into a separate directory.

It can be addressed in another pr.

Reply:
yes, I am analyzing its requirements about it to get a better interface, maybe more thinking when adding other formats.  put it here tmp since it is simple for now.

Reply:
for POST, parse the 'query' parameter first in case it is "insert ...  format".
if not (parse fail or not insert format ),  append the body and parse again.